### PR TITLE
Implementation of DatabaseSchemaStatus update

### PR DESCRIPTION
**Pull request by tinkaalgogoi**

### CODE REVIEW

File: atlas-db-controller/controller.go [hunk 1]
```diff
@@ -295,22 +297,28 @@ func (c *Controller) syncSchema(key string) error {
 	dbName := schema.Spec.Database
 	db, err := c.dbsLister.Databases(namespace).Get(dbName)
 	if err != nil {
-		runtime.HandleError(fmt.Errorf("cannot get database `%s`: %s", dbName, err))
+		schemaStatusMsg = fmt.Sprintf("cannot get database `%s`: %s", dbName, err)
```
Comment (jkraj) [inline atlas-db-controller/controller.go]: cannot get database info / failed to fetch database info

File: atlas-db-controller/controller.go [hunk 2]
```diff
@@ -295,22 +297,28 @@ func (c *Controller) syncSchema(key string) error {
...
 		return err
 	}
 
 	glog.Info(db.Spec.ServerType)
 	glog.Info(db.Spec.Dsn)
 
 	if db.Spec.ServerType != "postgres" { //  && db.Spec.ServerType != ...
-		err = fmt.Errorf("unsupported database server type `%s`", db.Spec.ServerType)
+		schemaStatusMsg = fmt.Sprintf("unsupported database server type `%s`", db.Spec.ServerType)
+		c.updateDatabaseSchemaStatus(key, schema, StateError, schemaStatusMsg)
```
Comment (jkraj) [inline atlas-db-controller/controller.go]: Can you add supported list as well.

File: atlas-db-controller/controller.go [hunk 3]
```diff
@@ -295,22 +297,28 @@ func (c *Controller) syncSchema(key string) error {
...
+		err = fmt.Errorf(schemaStatusMsg)
 		runtime.HandleError(err)
 		return err
 	}
 
 	mgrt, err := migrate.New(schema.Spec.Git, db.Spec.Dsn)
 	if err != nil {
-		err = fmt.Errorf("cannot create migrate: %s", err)
+		schemaStatusMsg = fmt.Sprintf("cannot create migrate: %s", err)
```
Comment (jkraj) [inline atlas-db-controller/controller.go]: cannot create migrate object / failed to initialize migrate engine

File: atlas-db-controller/controller.go [hunk 4]
```diff
@@ -320,33 +328,73 @@ func (c *Controller) syncSchema(key string) error {
...
 		glog.Infof("database `%s` has synced version %d", dbName, toVersion)
+		schemaStatusMsg = fmt.Sprintf("database `%s` has synced version %d", dbName, toVersion)
+		c.updateDatabaseSchemaStatus(key, schema, StateSuccess, schemaStatusMsg)
 		return nil
 	}
 
 	err = mgrt.Migrate(toVersion)
 	if err != nil {
-		err = fmt.Errorf("cannot migrate: %s", err)
+		schemaStatusMsg = fmt.Sprintf("cannot migrate: %s", err)
```
Comment (jkraj) [inline atlas-db-controller/controller.go]: add database name as well.

File: atlas-db-controller/controller.go (hunk 2 above)
Comment (tinkaalgogoi) [reply atlas-db-controller/controller.go]: right now it just doing a string check with postgres. Need to reimplement this with list of supported type dbs

**Pull request merged**